src/main/scala/nl/biopet/tools/bamstats/FlagStats.scala

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2014 Biopet
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package nl.biopet.tools.bamstats
+
+import java.io.{File, PrintWriter}
+import java.util.Locale
+
+import htsjdk.samtools.SAMRecord
+import nl.biopet.utils.conversions
+import play.api.libs.json.Json
+
+import scala.collection.mutable
+import util.Properties.lineSeparator
+
+class FlagStats {
+
+  private val flagStats: mutable.Map[FlagMethods.Value, Long] =
```

Comment:
Internal representation should be `Array[Long]`. Maps are to slow for each lookup.
By calling FlagMethods.zipWithIndex it can be accessed a lot faster.
This actions will be executed numbers of read * flagstats methods. In some cases this can exceed the 1.000.000.000